src/regex_impl.cc

```diff
@@ -81,7 +89,8 @@ struct ParsedRegex
         NodeIndex children_end;
         Codepoint value;
         Quantifier quantifier;
-    };
+        uint16_t filler = 0;
+    } __packed;
```

Comment:
`[[gnu::packed]]` should do the trick and does not necessitate the conditional macros as unrecognized attributes must be ignored.

Reply:
Thanks! I must admit I may have not quite kept up-to-speed with the C++ standards in the last, hm, twenty years... thanks for pointing the attribute syntax out to me!

I just pushed another commit to the branch that minimizes the changes. Let me know if you'd like me to squash/force-push a single commit instead.